pkg/blockbuilder/scheduler/scheduler.go

```diff
@@ -4,27 +4,34 @@ package scheduler
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"sync"
 	"time"
 
 	"github.com/go-kit/log"
 	"github.com/go-kit/log/level"
 	"github.com/grafana/dskit/services"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/twmb/franz-go/pkg/kadm"
-	"github.com/twmb/franz-go/pkg/kgo"
 
+	"github.com/grafana/mimir/pkg/blockbuilder"
 	"github.com/grafana/mimir/pkg/storage/ingest"
 )
 
 type BlockBuilderScheduler struct {
 	services.Service
 
-	kafkaClient *kgo.Client
+	adminClient *kadm.Client
+	jobs        *jobQueue
 	cfg         Config
 	logger      log.Logger
 	register    prometheus.Registerer
 	metrics     schedulerMetrics
+
+	mu        sync.Mutex
+	committed kadm.Offsets
+	dirty     bool
```

Comment:
To double-check: these fields are for the future updates, when the scheduler will take over managing of the commits, right? Maybe a todo comment here could clarify it?

Reply:
Thanks, that's correct. Will do!

Reply:
I've just removed these bits as they're entirely half-baked noise in this PR.